addons/mail/models/mail_channel.py

```diff
@@ -812,21 +801,16 @@ def channel_info(self):
                 # avoid sending potentially a lot of members for big channels
                 # exclude chat and other small channels from this optimization because they are
                 # assumed to be smaller and it's important to know the member list for them
-                info['members'] = sorted(list(channel._channel_info_format_member(member.partner_id, partner_format_by_partner[member.partner_id]) for member in members_by_channel[channel] if member.partner_id), key=lambda p: p['id'])
+                channel_data['channelMembers'] = [('insert', list(members_by_channel[channel]._mail_channel_member_format().values()))]
                 info['seen_partners_info'] = sorted([{
                     'id': cp.id,
                     'partner_id': cp.partner_id.id,
                     'fetched_message_id': cp.fetched_message_id.id,
                     'seen_message_id': cp.seen_message_id.id,
                 } for cp in members_by_channel[channel] if cp.partner_id], key=lambda p: p['partner_id'])
-                info['guestMembers'] = [('insert', sorted([{
-                    'id': member.guest_id.id,
-                    'name': member.guest_id.name,
-                } for member in members_by_channel[channel] if member.guest_id], key=lambda g: g['id']))]
-
             # add RTC sessions info
             info.update({
-                'invitedMembers': [('insert', invited_members_by_channel[channel].mail_channel_member_format())],
+                'invitedMembers': [('insert', list(invited_members_by_channel[channel]._mail_channel_member_format(fields={'id': True, 'channel': {}, 'persona': {'partner': {'id', 'name', 'im_status'}, 'guest': {'id', 'name', 'im_status'}}}).values()))],
```

Comment:
Do we still need the `insert` command in master ?

Reply:
for x2one we don't need them, for x2many it depends if we want to just insert an extra record or replace all existing records (it is sometimes done but it is prone to race conditions especially for quickly changing values such as a pending call)